### PR TITLE
Implement REST api that can create and get tasks saved in the database

### PR DESCRIPTION
**Pull request by mads256h**
Axum is used for the web framework and sqlx is used for connecting to the database.
The database is a sqlite database for simplicity.

The folder migrations contains sql commands that setup a sql database from scratch.
It is also used to upgrade an older sql database schema to a newer one. Sqlx keeps track of the schema version the database uses.

To build you need some one-time setup:
First create a file named .env in the root of the project. This contains the connection string used for connecting to the database. It can look like this:
```
DATABASE_URL=sqlite://dev.db
```
Then you need to install the sqlx commandline tool. Use it to create the database and run the available migrations.
```
cargo install sqlx-cli
cargo sqlx db create
cargo sqlx migrate run
```

### CODE REVIEW

File: src/data_model/event.rs [hunk 1]
```diff
@@ -1,9 +1,9 @@
+use ::chrono::{DateTime, Local};
```
Comment (S3JER) [inline src/data_model/event.rs:1]: huh?

**Review by S3JER — CHANGES_REQUESTED** [review]

**Review by S3JER — APPROVED** [review]

**Comment by mads256h** [comment]
You can test the create api with
```
curl --header "Content-Type: application/json"   --request POST   --data '{"id":-1,"timespan":{"start": 101, "end": 110},"duration":5,"effect":5000}'   http://localhost:3000/create
```

And open the browser on localhost:3000/ to see the created tasks

File: src/data_model/task.rs [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-}
+use serde::{Deserialize, Serialize};
 
-struct Task {
-    device_id: Uuid,
-    effect: i16,
-    duration: i32,
-    interval: TimeInteval,
+#[derive(Serialize, Deserialize)]
+pub struct Timespan {
```
Comment (henneboy) [inline src/data_model/task.rs:4]:
```suggestion
pub struct TimePeriod {
```
Rename to TimePeriod or something else, TimeSpan is https://learn.microsoft.com/en-us/dotnet/api/system.timespan?view=net-8.0, plz don't use that name

File: src/data_model/task.rs [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
 
-pub fn hello() {
-    println!("Hello World");
-}
\ No newline at end of file
+#[derive(Serialize, Deserialize)]
+pub struct Task {
+    pub id: i64,
+    pub timespan: Timespan,
+    pub duration: i64,
```
Comment (henneboy) [inline src/data_model/task.rs:13]:
```suggestion
    pub durationInMinuttes: i64,
```
Add the unit in the name or create a data type for it.

**Review by henneboy — CHANGES_REQUESTED** [review]

File: src/data_model/task.rs (hunk 2 above)
Comment (mads256h) [reply src/data_model/task.rs:4]: This is exactly the same

**Review by henneboy — APPROVED** [review]

**Pull request merged**